validator/project_validator.go

```diff
@@ -1829,14 +1829,16 @@ func validateTVDependsOnTV(dependentTask, dependedOnTask model.TVPair, statuses
 			}
 		}
 
-		if fromTaskUnit.RunsOnPatches() && (!toTaskUnit.RunsOnPatches() || edgeInfo.PatchOptional) {
+		// PatchOptional dependencies are skipped when the edge.From task is running on a patch.
```

Comment:
preference for using the variable name (fromTaskUnit) rather than the struct name (edge.From) in comments, since that requires another translation step (and it's easy to read the . as the end of a sentence when skimming)

Reply:
done.
I'd actually meant the variable from lines 1822-1823 but I think that further underscores your point 😃